src/id-token/IDToken.ts

```diff
@@ -111,30 +151,23 @@ export class IDToken {
   public async verify(verifyOpts: VerifyAuthorizationResponseOpts): Promise<VerifiedIDToken> {
     assertValidVerifyOpts(verifyOpts);
 
-    const { header, payload } = parseJWT(await this.jwt());
-    this.assertValidResponseJWT({ header, payload });
+    const parsedJwt = parseJWT(this._jwt);
```

Comment:
Okay, so here's the problem with leaving it as it was.
The method this.jwt() creates the jwt if it hasn't been created yet, which I find strange because that means we're verifying a jwt that we previously created. We can keep the current behaviour, but then how do we create the jwt? Do we sign it with the custom method? Passing a JwtIssuer to the verification method?
For now I changed the implementation that it throws an Error if this._jwt is not set. Not sure what you would prefer here?

Reply:
The whole idea is to do a lazy evaluation. Be aware that the payloads can be used both on the holder side as well as on the RP side. So whenever you already had a JWT string, you could use the class to pass that in and get the accompanying decoded payload. But as a RP you could also build the request object first and then convert it into a JWT